src/Microsoft.OData.Core/Batch/ODataBatchOperationMessage.cs

```diff
@@ -141,13 +141,22 @@ public override Task<Stream> GetStreamAsync()
         {
             this.VerifyNotCompleted();
 
-            // notify the listener that the stream has been requested
-            Task listenerTask = this.operationListener.StreamRequestedAsync();
+            return GetStreamInnerAsync();
 
-            // now remember that we are done processing the part header data (and only the payload is missing)
-            Stream contentStream = this.contentStreamCreatorFunc();
-            this.PartHeaderProcessingCompleted();
-            return listenerTask.FollowOnSuccessWith(task => { return (Stream)contentStream; });
+            async Task<Stream> GetStreamInnerAsync()

```

Comment:
is this method being used anywhere else? is there a reason for introducing it or it is just for organizational purposes?

Reply:
It's a local function. The reason for introducing it is to avoid adding the async modifier to the method since that introduces a small diff to the API.